lib/fabrik/VirtualHostInstance.js

```diff
@@ -0,0 +1,93 @@
+const _ = require('lodash');
+const Promise = require('bluebird');
+const BaseInstance = require('./BaseInstance');
+const bosh = require('../bosh');
+const utils = require('../utils');
+const mapper = require('./VirtualHostRelationMapper');
+const catalog = require('../models').catalog;
+
+class VirtualHostInstance extends BaseInstance {
+
+  constructor(guid, manager) {
+    super(guid, manager);
+    this.director = bosh.director;
+    this.mapper = mapper.VirtualHostRelationMapper;
+  }
+
+  get async() {
+    return false;
+  }
+
+  initialize() {
+    return this.getDeploymentNameForVirtualHostInstanceId(this.guid)
+      .then((deploymentName) => this.deploymentName = deploymentName);
+  }
+
+  create(params) {
+    const rabbitmqInstanceName = params.parameters.dedicated_rabbitmq_instance;
+    return this.cloudController.getMultiTenantServiceInstanceIdWithName(rabbitmqInstanceName, params.organization_guid, params.space_guid, 'rabbitmq')
+      .then((serviceInstanceId) => {
+        return this.director.getDeploymentNameForInstanceId(serviceInstanceId);
+      })
```

Comment:
unnecesary return when there is just one line Can just inline the code like this and have the promise chain. 
```
this.cloudController.getMultiTenantServiceInstanceIdWithName(rabbitmqInstanceName, params.organization_guid, params.space_guid, 'rabbitmq')
 .then(serviceInstanceId => this.director.getDeploymentNameForInstanceId(serviceInstanceId)
 .then(deploymentName => this.manager.createOrUpdateVirtualHost(deploymentName, params, this.guid));
```